rust/src/array.rs

```diff
@@ -24,7 +24,7 @@ use std::sync::Arc;
 
 use array_data::*;
 use buffer::*;
```

Comment:
I there is a need to use the `use <module>::*` pattern for `builder` and `datatypes` (and `array`) because they define a lot of types, but maybe we should avoid this for `array_data` and `buffer`?